src/R/Components/Impl/Plots/Implementation/RPlotManager.cs

```diff
@@ -24,12 +24,16 @@ internal class RPlotManager : IRPlotManager {
         private readonly IRInteractiveWorkflow _interactiveWorkflow;
         private readonly IFileSystem _fileSystem;
         private readonly DisposableBag _disposableBag;
+        private readonly object _devicesLock = new object();
         private readonly List<IRPlotDevice> _devices = new List<IRPlotDevice>();
         private readonly Dictionary<int, IRPlotDeviceVisualComponent> _visualComponents = new Dictionary<int, IRPlotDeviceVisualComponent>();
         private readonly Dictionary<Guid, IRPlotDeviceVisualComponent> _assignedVisualComponents = new Dictionary<Guid, IRPlotDeviceVisualComponent>();
         private readonly List<IRPlotDeviceVisualComponent> _unassignedVisualComponents = new List<IRPlotDeviceVisualComponent>();
         private readonly ICoreShell _shell;
 
+        private TaskCompletionSource<LocatorResult> _locatorTcs;
+        private CancellationTokenRegistration? _locatorCancelTokenRegistration;
```

Comment:
I don't think you really need to make it nullable. It isn't checked for null, and `Dispose` can be called on `CancellationTokenRegistration` multiple times.